manim/utils/tex_file_writing.py

```diff
@@ -21,21 +21,56 @@ def tex_hash(expression):
     return hasher.hexdigest()[:16]
 
 
-def tex_to_svg_file(expression, source_type, tex_template=None):
+def tex_to_svg_file(expression, environment=None, tex_template=None):
+    """Takes a tex expression and returns the svg version of the compiled tex
+
+    Parameters
+    ----------
+    expression : :class:`str`
+        String containing the TeX expression to be rendered, e.g. ``\\sqrt{2}`` or ``foo``
+    environment : Optional[:class:`str`], optional
+        The string containing the environment in which the expression should be typeset, e.g. ``align*``
+    tex_template : Optional[:class:`TexTemplate`], optional
+        Template class used to typesetting. If not set, use default template set via `config["tex_template"]`
+
+    Returns
+    -------
+    :class:`str`
+        Path to generated SVG file.
+    """
     if tex_template is None:
         tex_template = config["tex_template"]
-    tex_file = generate_tex_file(expression, tex_template, source_type)
-    dvi_file = tex_to_dvi(tex_file, tex_template.use_ctex, tex_template.tex_compiler)
-    return dvi_to_svg(
-        dvi_file, use_ctex=tex_template.use_ctex, tex_compiler=tex_template.tex_compiler
+    tex_file = generate_tex_file(expression, environment, tex_template)
+    dvi_file = compile_tex(
+        tex_file, tex_template.tex_compiler, tex_template.output_format
     )
-
-
-def generate_tex_file(expression, tex_template, source_type):
-    if source_type == "text":
-        output = tex_template.get_text_for_text_mode(expression)
-    elif source_type == "tex":
-        output = tex_template.get_text_for_tex_mode(expression)
+    return convert_to_svg(dvi_file, tex_template.output_format)
+
+
+def generate_tex_file(expression, environment=None, tex_template=None):
+    """Takes a tex expression (and an optional tex environment),
+    and returns a fully formed tex file ready for compilation.
+
+    Parameters
+    ----------
+    expression : :class:`str`
+        String containing the TeX expression to be rendered, e.g. ``\\sqrt{2}`` or ``foo``
+    environment : Optional[:class:`str`], optional
+        The string containing the environment in which the expression should be typeset, e.g. ``align*``
+    tex_template : Optional[:class:`TexTemplate`], optional
```

Comment:
TexTemplate is in another file, so we must tell sphinx to search for it using `.`; `~` to shorten to `TexTemplate` instead of the full path
```suggestion
    tex_template : Optional[:class:`~.TexTemplate`], optional
```